docs/extensions/installer-extensions.md

```diff
@@ -0,0 +1,217 @@
+    category: prediction-serving
+    description: kserve prediction service API
+    endpoints:
+      - url: http://kubernetes.default.svc
+        type: internal
+  - id: UI
+    resource: kserve-ui
+    category: UI
+    description: KServe UI
+gateways:
+  - name: kserve-web-app
+    namespace: kserve
+    servicehost: kserve-models-web-app
+    port: 80
+rolerules:
+  - apigroups:
+      - serving.kserve.io
+    resources:
+      - inferenceservices
+    verbs:
+      - get
+      - list
+      - create
+      - patch
+      - watch
+```
+
+The descriptor specifies a list of installation and un-installation steps, such as Kubernetes manifests and Kustomize files, that have to be executed in order to deploy the application or to subsequently remove it from a Kubernetes cluster. The other fields in the descriptor are used to describe the application, its capabilities and its permission requirements.
+
+The extension descriptor is broken down into sections and explored piece by piece in the following paragraphs.
+
+```yaml
+name: kserve
+product: kserve
+version: "0.7.0"
+description: |
+  Serverless Inferencing on Kubernetes
+[...]
+```
+
+This part is self explanatory. The `name` field is the extension identifier that along with the `product`, `version` and `description` fields is used to populate the [Extension Record](extension-registry.md#extension-records) that FuseML will use to register KServe as an extension in the [FuseML Extension Registry](extension-registry.md) after installation.
+
+```yaml
+[...]
+requires:
+  - knative
+  - cert-manager
+[...]
+```
+
+Extensions can depend on other extensions. If a description file contains a `requires` field, the value is expected to be a list of names of other extensions that are considered requirements for the current one. The FuseML installer will take care to install all dependencies in the right order, so they do not need to be explicitly listed on the command line.
+
+```yaml
+[...]
+install:
+  - type: manifest
+    location: https://github.com/kserve/kserve/releases/download/v0.7.0/kserve.yaml
+    waitfor:
+      - namespace: kserve
+        selector: control-plane=kserve-controller-manager
+  - type: kustomize
+    location: kustomize
+    waitfor:
+      - namespace: kserve
+        selector: app.kubernetes.io/component=kserve-models-web-app
+uninstall:
+  - type: kustomize
+    location: kustomize
+  - type: manifest
+    location: https://github.com/kserve/kserve/releases/download/v0.7.0/kserve.yaml
+[...]
+```
+
+The `install` and `uninstall` sections list the steps necessary to install, configure and uninstall the extension. KServe is installed through a combination of Kubernetes manifests and Kustomize files. The following types of installation mechanisms are currently supported:
+
+- `helm` - helm chart
+- `manifest` - plain Kubernetes manifest, to be installed using `kubectl`. All information (including namespace for example) is expected to be present in the manifest file.
```

Comment:
That namespace example should probably be removed - if a namespace is not part of a manifest, fuseml-installer will create it if is is part of description file

Reply:
I removed it